src/rp2040/hardware_structs/include/hardware/structs/dma.h

```diff
@@ -8,46 +10,175 @@
 #define _HARDWARE_STRUCTS_DMA_H
 
 #include "hardware/address_mapped.h"
-#include "hardware/platform_defs.h"
 #include "hardware/regs/dma.h"
 
+// reference to datasheet: https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf#tab-registerlist_dma
+// BITMASK : FIELDNAME [BITRANGE] (RESETVALUE): DESCRIPTION
+
 typedef struct {
+    _REG_(DMA_CH0_READ_ADDR_OFFSET)
+    // DMA Channel 0 Read Address pointer
     io_rw_32 read_addr;
+
+    _REG_(DMA_CH0_WRITE_ADDR_OFFSET)
+    // DMA Channel 0 Write Address pointer
     io_rw_32 write_addr;
+
+    _REG_(DMA_CH0_TRANS_COUNT_OFFSET)
+    // DMA Channel 0 Transfer Count
     io_rw_32 transfer_count;
+
+    _REG_(DMA_CH0_CTRL_TRIG_OFFSET)
+    // DMA Channel 0 Control and Status
+    // 0x80000000 [31]    : AHB_ERROR (0): Logical OR of the READ_ERROR and WRITE_ERROR flags
+    // 0x40000000 [30]    : READ_ERROR (0): If 1, the channel received a read bus error
+    // 0x20000000 [29]    : WRITE_ERROR (0): If 1, the channel received a write bus error
+    // 0x01000000 [24]    : BUSY (0): This flag goes high when the channel starts a new transfer...
+    // 0x00800000 [23]    : SNIFF_EN (0): If 1, this channel's data transfers are visible to the sniff...
+    // 0x00400000 [22]    : BSWAP (0): Apply byte-swap transformation to DMA data
+    // 0x00200000 [21]    : IRQ_QUIET (0): In QUIET mode, the channel does not generate IRQs at the end...
+    // 0x001f8000 [15-20] : TREQ_SEL (0): Select a Transfer Request signal
+    // 0x00007800 [11-14] : CHAIN_TO (0): When this channel completes, it will trigger the channel...
+    // 0x00000400 [10]    : RING_SEL (0): Select whether RING_SIZE applies to read or write addresses
+    // 0x000003c0 [6-9]   : RING_SIZE (0): Size of address wrap region
+    // 0x00000020 [5]     : INCR_WRITE (0): If 1, the write address increments with each transfer
+    // 0x00000010 [4]     : INCR_READ (0): If 1, the read address increments with each transfer
+    // 0x0000000c [2-3]   : DATA_SIZE (0): Set the size of each bus transfer (byte/halfword/word)
+    // 0x00000002 [1]     : HIGH_PRIORITY (0): HIGH_PRIORITY gives a channel preferential treatment in...
+    // 0x00000001 [0]     : EN (0): DMA Channel Enable
     io_rw_32 ctrl_trig;
-    io_rw_32 al1_ctrl;
-    io_rw_32 al1_read_addr;
-    io_rw_32 al1_write_addr;
-    io_rw_32 al1_transfer_count_trig;
-    io_rw_32 al2_ctrl;
-    io_rw_32 al2_transfer_count;
-    io_rw_32 al2_read_addr;
-    io_rw_32 al2_write_addr_trig;
-    io_rw_32 al3_ctrl;
-    io_rw_32 al3_write_addr;
-    io_rw_32 al3_transfer_count;
-    io_rw_32 al3_read_addr_trig;
-}  dma_channel_hw_t;
+
+    _REG_(DMA_CH0_AL1_CTRL_OFFSET)
+    // Alias for channel 0 CTRL register
+    io_ro_32 al1_ctrl;
```

Comment:
It looks weird for a "control" register to be read-only? But looking at the [datasheet](https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf) on page 115, it indeed says that `CH0_AL1_CTRL` is "RO". Is that right @Wren6991 or should it be "RW"? :man_shrugging:

Reply:
Looks like this is actually causing the build of this PR to fail.

Reply:
Ah, yes. I noticed this too

Reply:
See also https://github.com/raspberrypi/pico-feedback/issues/190

Reply:
this does not seemed to have been fixed in the latest SVD... they should all be R/W